hpycc/connection.py

```diff
@@ -265,18 +274,22 @@ def run_ecl_script(self, script, syntax_check, delete_workunit,
 
         try:
             result = self._run_command(base_cmd)
+            # TODO dont like this nested try except
         except subprocess.SubprocessError as e:
             if delete_workunit:
-                wuid = parse_wuid_from_failed_response(e.args[0].decode())
-                delete.delete_workunit(self, wuid)
+                try:
+                    wuid = parse_wuid_from_failed_response(e.args[0].decode())
+                    delete.delete_workunit(self, wuid)
+                except AttributeError as e2:
+                    print('Job failed and unable to delete workunit, likely a connection error. Response: %s' % e2 )
             raise e
         else:
             if delete_workunit:
                 wuid = parse_wuid_from_xml(result.stdout)
                 delete.delete_workunit(self, wuid)
             return result
 
-    def run_url_request(self, url, max_attempts, max_sleep):
+    def run_url_request(self, url, max_attempts, max_sleep, min_sleep):
```

Comment:
min_sleep was created as, in the case of larger files, you may need a longer timeout to prevent repeated failures. We could make min_sleep a function of max sleep I suppose?

Reply:
Personally I would do something like that, but I'm aware it might be messy/ difficult to implement in a clever way. At the very least we should do some checking to make sure max_sleep > min_sleep.

Reply:
For now just done the test as discussed.